src/main/java/core/basesyntax/service/impl/SupplyTransactionHandler.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Database;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.service.TransactionHandler;
+import java.math.BigInteger;
+
+public class SupplyTransactionHandler implements TransactionHandler {
+    @Override
+    public void handle(Database database, FruitType fruitType, BigInteger quantity) {
+        if (quantity == null || quantity.signum() < 0) {
+            throw new IllegalArgumentException("Quantity is wrong. Cannot be: " + quantity);
```

The quantity validation throws `IllegalArgumentException` with a short message. Per checklist guidance, prefer informative messages that include context (which fruit and which operation). Quote: "Don't ignore exceptions in try-catch blocks" / rethrow informative RuntimeExceptions. Improve message to something like: `throw new RuntimeException("Invalid supply quantity for fruit='" + fruitType + "': " + quantity)` so debugging input errors is easier.